keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/binary/DoubleBinaryOpVertex.java

```diff
@@ -67,7 +68,7 @@ public PartialDerivatives forwardModeAutoDifferentiation(Map<Vertex, PartialDeri
         try {
             return forwardModeAutoDifferentiation(derivativeOfParentsWithRespectToInputs.get(left), derivativeOfParentsWithRespectToInputs.get(right));
         } catch (UnsupportedOperationException e) {
-            return super.forwardModeAutoDifferentiation(derivativeOfParentsWithRespectToInputs);
+            return Differentiable.super.forwardModeAutoDifferentiation(derivativeOfParentsWithRespectToInputs);
```

Comment:
Differentiable.super? I think this is another interface DifferentiableBinaryOp.

Reply:
So the super used to lead to the `DoubleVertex`'s implementation of `forwardModeAutoDifferentiation`, which I moved to `Differentiable` as a default. Why would we need another interface for it?